R/content.R

```diff
@@ -111,7 +111,26 @@ Content <- R6::R6Class(
           x
         })
       }
-      purrr::map(parsed, ~ purrr::list_modify(.x, app_guid = self$content$guid))
+      # Ensure content identifiers are included.
+      # The `app_guid` field was never returned by Connect, but is added here
+      # for backward compatibility with older `connectapi` versions, which added
+      # a field named `app_guid` to job data.
+      if (compare_connect_version(self$connect$version, "2025.01.0") < 0) {
+        # Versions of Connect below 2025.01.0 only included `app_id`. We must
+        # add all other fields.
+        purrr::map(parsed, ~ purrr::list_modify(.x,
+          content_id = .x$app_id,
+          app_guid = self$content$guid,
+          content_guid = self$content_guid
+        ))
```

Comment:
Although this logic would work by itself, I added the `if` statement here to make clearer what data we're expecting from the server in different conditions.